tests/x11/yast2_snapper.pm

```diff
@@ -0,0 +1,92 @@
+use base "x11test";
+use testapi;
+
+# Test for basic yast2-snapper functionality. It assumes the data of the
+# opensuse distri to be available at /home/$username/data (as granted by
+# console_setup.pm)
+
+# Helper for letting y2-snapper to create a snapper snapshot
+sub y2snapper_create_snapshot() {
+    my $self = shift;
+    my $name = shift || "Awesome Snapshot";
+   # Open the 'C'reate dialog and wait until it is there
+    send_key "alt-c";
+    assert_screen 'yast2_snapper-createsnapshotdialog', 100;
+    # Fill the form and finish by pressing the 'O'k-button
+    type_string $name;
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    type_string "a=1,b=2";
+    sleep 2;
+    send_key "alt-o";
+}
+sub run() {
+    my $self = shift;
+    # Make sure yast2-snapper is installed (if not: install it)
+    ensure_installed "yast2-snapper";
+    # Start an xterm as root
+    x11_start_program("xterm");
+    wait_idle;
+    become_root;
+    # Create a snapper config
+    script_run "btrfs subvolume create /tmp/testdata";
+    script_run "snapper create-config /tmp/testdata";
```

Comment:
for any btrfs test, you should be using the HDDSIZEGB variable to make
openQA give your VM a larger hard drive

Source:
https://github.com/os-autoinst/openQA/blob/master/docs/GettingStarted.asciidoc

I think testing an extra subvolume isn't a terrible way to go, but I would
prefer if we can alter this test to poke around the default config a little
better

On 6 June 2015 at 10:01, Christopher Hofmann <notifications@github.com>
wrote:

> In tests/x11/yast2_snapper.pm
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/482#discussion_r31865875>
> :
>
> > +    sleep 2;
> > +    type_string "a=1,b=2";
> > +    sleep 2;
> > +    send_key "alt-o";
> > +}
> > +sub run() {
> > +    my $self = shift;
> > +    # Make sure yast2-snapper is installed (if not: install it)
> > +    ensure_installed "yast2-snapper";
> > +    # Start an xterm as root
> > +    x11_start_program("xterm");
> > +    wait_idle;
> > +    become_root;
> > +    # Create a snapper config
> > +    script_run "btrfs subvolume create /tmp/testdata";
> > +    script_run "snapper create-config /tmp/testdata";
>
> Since snapper is deactivated on small harddrives by default it is actually
> not enabled because of our current openQA settings. so I have to create a
> snapper config anyway.
> I decided to go to an extra subvolume to be sure that I only see the
> changes I do. Just using /tmp leads to some extra changes because yast
> creates some files there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/482/files#r31865875>
> .
>